pkg/nmstate/funcs.go

```diff
@@ -2,22 +2,26 @@ package nmstate
 
 import (
 	nmstateshared "github.com/nmstate/kubernetes-nmstate/api/shared"
+
 	"github.com/tidwall/gjson"
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/yaml"
 )
 
-// GetDesiredStatedBridgeName - Get the bridge name associated with the desiredState of the attachConfiguration
-func GetDesiredStatedBridgeName(desiredStateBytes []byte) (string, error) {
+// GetDesiredStateBridgeName - Get the name associated with the desiredState bridge interface
+func GetDesiredStateBridgeName(desiredStateBytes []byte) (string, error) {
 	bridge := ""
 
 	jsonStr, err := GetDesiredStateAsString(desiredStateBytes)
 	if err != nil {
 		return "", err
 	}
 
-	if gjson.Get(jsonStr, "interfaces.#.name").Exists() {
-		bridge = gjson.Get(jsonStr, "interfaces.#.name").Array()[0].String()
+	if gjson.Get(jsonStr, "interfaces.#.bridge").Exists() &&
+		gjson.Get(jsonStr, "interfaces.#.name").Exists() &&
+		len(gjson.Get(jsonStr, `interfaces.#(bridge).state`).Array()) > 0 {
```

Comment:
Should this check for `name` instead of `state`?

Reply:
yes indeed, well spotted